identity-apps-core/components/org.wso2.identity.apps.common/src/test/java/org/wso2/identity/apps/common/listener/AppPortalRoleManagementListenerTest.java

```diff
@@ -0,0 +1,201 @@
+                                String tenantDomain, boolean isAdminRole, boolean isOrganization, String adminUserId)
+        throws Exception {
+
+        AppPortalRoleManagementListener appPortalRoleManagementListener = spy(
+            new AppPortalRoleManagementListener(true));
+
+        // Set up the mock behaviors
+        organizationManagementUtil.when(() ->
+            OrganizationManagementUtil.isOrganization(tenantDomain)).thenReturn(isOrganization);
+        AbstractUserStoreManager mockUserStoreManager = mock(AbstractUserStoreManager.class);
+        when(mockUserRealm.getRealmConfiguration()).thenReturn(mock(RealmConfiguration.class));
+        when(mockUserRealm.getRealmConfiguration().getAdminUserName()).thenReturn(ADMINISTRATOR);
+        when(mockUserRealm.getUserStoreManager()).thenReturn(mockUserStoreManager);
+        when(mockUserStoreManager.getUserIDFromUserName(anyString()))
+            .thenReturn(adminUserId);
+        mockAppsCommonDataHolder(appsCommonDataHolder);
+        when(roleManagementService.getRole(roleId, tenantDomain)).thenReturn(role);
+        setRoleAttributes(isAdminRole);
+
+        // Call the method
+        if (!isAdminRole || deletedUserIDList == null || isOrganization || !deletedUserIDList.contains(adminUserId)) {
+            appPortalRoleManagementListener.preUpdateUserListOfRole(roleId, newUserIDList, deletedUserIDList,
+                tenantDomain);
+        } else {
+            Exception exception = null;
+            try {
+                appPortalRoleManagementListener.preUpdateUserListOfRole(roleId, newUserIDList, deletedUserIDList,
+                    tenantDomain);
+                fail("Expected IdentityRoleManagementException was not thrown");
+            } catch (IdentityRoleManagementException e) {
+                exception = e;
+            }
+            assertNotNull(exception);
+            assertEquals(exception.getMessage(), "Deleting the tenant admin from 'Administrator' role " +
+                "belongs to the 'Console' application is not allowed.");
+        }
+
+        if (deletedUserIDList == null || !isAdminRole) {
+            organizationManagementUtil.verify(() -> OrganizationManagementUtil.isOrganization(tenantDomain), never());
+        } else if (isOrganization) {
+            organizationManagementUtil.verify(() -> OrganizationManagementUtil.isOrganization(tenantDomain));
+        }
+    }
+
+    private void setRoleAttributes(boolean isAdminRole) {
+
+        if (isAdminRole) {
+            when(role.getName()).thenReturn(ADMINISTRATOR);
+            when(role.getAudience()).thenReturn(APPLICATION);
+            when(role.getAudienceName()).thenReturn(CONSOLE_APP);
+        } else {
+            when(role.getName()).thenReturn(ROLE_NAME);
+            when(role.getAudience()).thenReturn(ORGANIZATION);
+            when(role.getAudienceName()).thenReturn("App1");
```

Comment:
```suggestion
            when(role.getAudienceName()).thenReturn("org1");
```
Shall we change the audience name matching to audience 

or changet the audince the application if you want to keep audience name as "App1"

Reply:
Changed this to `Org1`.